### PR TITLE
Match typing docs - show use of annotations with ellipsis (...)

### PR DESCRIPTION
**Pull request by maueroats**
Update of #1057

### CODE REVIEW

File: typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl [hunk 1]
```diff
@@ -244,7 +244,17 @@ anticipate with their pattern matching, e.g.:
   (eval:error
    (size 42))]
 
+Patterns involving an ellipsis @racket[...] for repetition may generate a 
+@racket[for] loop that requires annotations on variables to type check. 
+The (deliberately obscure) code below does not type check without the type 
+annotation on the match pattern variable @racket[c].
 
+@examples[#:label #f #:eval the-eval
+  (: without-10 : (Listof Integer) -> (Listof Integer))
```
Comment (sorawee) [inline typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl:253]: The name `without-10` is kinda weird, now that the first element is not 10.

Comment (maueroats) [reply typed-racket-doc/typed-racket/scribblings/guide/caveats.scrbl]: Name improved to `skip-a-num`, which is kind of casual.
I also consided `rest-nums` and variations on it (`rest-nums-only`, `rest-numeric`, `rest-intonly`).
I don't have a real preference or guiding principle for the choice.

**Comment by samth** [comment]
Sorry, I wasn't quite clear on what I meant. Here's the kind of program I'm suggesting:

```
(: suml : (Listof Any) -> Number)
(define (suml l)
   (match l
     [(list (? number? #{c : (Listof Number)}) ...) (apply + c)]
     [_ 0]))
```

**Comment by maueroats** [comment]
This should be a draft until the code is tested with the recent typed racket fixes by @samth.

**Comment by samth** [comment]
I changed `match` so you should be able to see the original example working in the latest snapshot.

**Comment by samth** [comment]
Is this ready?

**Comment by maueroats** [comment]
"Real Work" is busy, I will ping you when this is clean. Soon.

**Comment by maueroats** [comment]
The following code runs in Racket but generates an error in Scribble. Any help understanding why? @samth 
```racket
@examples[#:label #f #:eval the-eval
  (: do-nothing (-> (Listof Integer) (Listof Integer)))
  (define (do-nothing lst)
    (match lst
      [(list (? number? #{c : (Listof Integer)}) ...)   c]))]
```

**Comment by samth** [comment]
The problem is that the typed/racket reader extends the racket reader to support that syntax. I think you probably want to use codeblock and specify the language explicitly.

**Comment by maueroats** [comment]
Replaced by #1070

**Pull request closed without merging**